x-pack/plugins/cloud_security_posture/public/components/csp_page_template.tsx

```diff
@@ -0,0 +1,175 @@
+
+interface QueryError {
+  body: {
+    error: string;
+    message: string;
+    statusCode: number;
+  };
+  name: string;
+}
+
+const activeItemStyle = { fontWeight: 700 };
+
+export const getSideNavItems = (
+  navigationItems: Record<string, CspNavigationItem>
+): NonNullable<KibanaPageTemplateProps['solutionNav']>['items'] =>
+  Object.entries(navigationItems)
+    .filter(([_, navigationItem]) => !navigationItem.disabled)
+    .map(([id, navigationItem]) => ({
+      id,
+      name: navigationItem.name,
+      renderItem: () => (
+        <NavLink to={navigationItem.path} activeStyle={activeItemStyle}>
+          {navigationItem.name}
+        </NavLink>
+      ),
+    }));
+
+const DEFAULT_PAGE_PROPS: KibanaPageTemplateProps = {
+  solutionNav: {
+    name: CLOUD_SECURITY_POSTURE,
+    items: getSideNavItems({
+      dashboard: allNavigationItems.dashboard,
+      findings: allNavigationItems.findings,
+      benchmark: allNavigationItems.benchmarks,
+    }),
+  },
+  restrictWidth: false,
+};
+
+export const DEFAULT_NO_DATA_CONFIG: KibanaPageTemplateProps['noDataConfig'] = {
+  pageTitle: DEFAULT_NO_DATA_TEXT.PAGE_TITLE,
+  solution: DEFAULT_NO_DATA_TEXT.SOLUTION,
+  // TODO: Add real docs link once we have it
+  docsLink: 'https://www.elastic.co/guide/index.html',
+  logo: 'logoSecurity',
+  actions: {},
+};
+
+const getPackageNotInstalledConfig = (
+  cisIntegrationLink: string
+): KibanaPageTemplateProps['noDataConfig'] => ({
+  pageTitle: PACKAGE_NOT_INSTALLED_TEXT.PAGE_TITLE,
+  solution: PACKAGE_NOT_INSTALLED_TEXT.SOLUTION,
+  // TODO: Add real docs link once we have it
+  docsLink: 'https://www.elastic.co/guide/index.html',
+  logo: 'logoSecurity',
+  actions: {
+    elasticAgent: {
+      href: cisIntegrationLink,
+      title: PACKAGE_NOT_INSTALLED_TEXT.BUTTON_TITLE,
+      description: PACKAGE_NOT_INSTALLED_TEXT.DESCRIPTION,
+    },
+  },
+});
+
+const DefaultLoading = () => <CspLoadingState>{LOADING}</CspLoadingState>;
+
+const DefaultError = (error: QueryError) => (
+  <EuiEmptyPrompt
+    color="danger"
+    iconType="alert"
+    title={
+      <>
+        <EuiTitle>
+          <h2>
+            <FormattedMessage
+              id="xpack.csp.pageTemplate.loadErrorMessage"
+              defaultMessage="We couldn't fetch your cloud security posture data"
+            />
+          </h2>
+        </EuiTitle>
+        {error.body?.error && error.body?.statusCode && (
+          <EuiTitle size="xs">
+            <h5>{`${error.body.error} (${error.body.statusCode})`}</h5>
+          </EuiTitle>
+        )}
+        {error.body?.message && (
+          <EuiTitle size="xs">
+            <h5>{error.body.message}</h5>
+          </EuiTitle>
+        )}
+      </>
+    }
+  />
+);
+
+export const CspPageTemplate: React.FC<
+  KibanaPageTemplateProps & {
+    loadingRender?: () => JSX.Element;
+    errorRender?: (error: QueryError) => JSX.Element;
+    query?: UseQueryResult;
```

Comment:
it's an implementation option that I agree is more versatile but it also more work need to be handled from the outside just in case we decide to dump react-query. I don't think we are going to dump it, and if we will, we can work to change this interface which shouldn't be too hard. for the time being, the current implementation saves time and hassle, and is very easy to use.